File: uva_1213.cpp
# include <iostream>
# include <vector>
# include <cmath>
#define MAXINPUT 1120
#define MAXKIND 14

using namespace std;
vector<int> prime_list(){
    vector<int> prime;
    for(int i=2; i<=MAXINPUT; ++i){
        bool check_prime=true;
        for(int j=2; j<=int(sqrt(i) ) && check_prime; ++j)
            if ( !(i%j) )
                check_prime=false;
        if (check_prime)
            prime.push_back(i);
    }
    return prime;
}
int main(void){

    vector<int> prime=prime_list();
    int ans[MAXINPUT+1][MAXKIND+1] = {0};

    ans[0][0] = 1;
    for (int p = 0; p < prime.size(); ++p) {
        for (int n = MAXINPUT; n >= prime[p]; --n) {
            for (int k = 1; k <= MAXKIND; ++k) {
                ans[n][k] += ans[n-prime[p]][k-1];
            }
        }
    }
 
    int n, k;
    while(cin >> n >> k){
        if(!n&&!k)
            break;
        else
            cout << ans[n][k] << endl;
    }

    return 0;
}

File: uva_1213.py
import math
MAXINPUT = 1120
MAXKIND = 14
def prime_list():
    prime=[]
    for i in range(2, MAXINPUT+1):
        check_prime=True
        for j in range(2, int(math.sqrt(i))+1 ):
            if not i%j:
                check_prime=False
                break
        if check_prime:
            prime.append(i)
    return prime

prime=prime_list()
ans=[ [0]*(MAXKIND+1) for _ in range(MAXINPUT+1)]

ans[0][0]=1
for p in prime:
    for n in range(MAXINPUT, p-1, -1):
        for k in range(1, MAXKIND+1):
            ans[n][k]+=ans[n-p][k-1]

while True:
    n, k=input().split()
    n, k=int(n), int(k)
    if not n and not k:
        break
    else:
        print(ans[n][k])